### PR TITLE
invoice list more options Bugs

### PR DESCRIPTION
**Pull request by Shruti-Apte**
### **Notion :**
NA
### **What :**
- More options updated on the invoice list page mobile view.
### **Why :**
- Previously it was linked to desktop modals.
### **Preview :**
Screenshot 2023-04-21 at 5 43 24 PM

### **Todos** (for testing):
- Needs to be tested thoroughly.

### CODE REVIEW

File: app/javascript/src/components/Invoices/Edit/index.tsx [hunk 1]
```diff
@@ -133,90 +137,124 @@ const EditInvoice = () => {
...
+            }}
           />
-          <div className="py-5 pl-10">
-            <InvoiceTable
+          <div className="m-0 mt-5 mb-10 w-full bg-miru-gray-100 p-0">
+            <CompanyInfo company={invoiceDetails.company} />
+            <InvoiceDetails
+              optionSelected
+              amount={amount}
+              clientList={invoiceDetails.companyClientList}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Edit/index.tsx:161]: destructure `invoiceDetails`

File: app/javascript/src/components/Invoices/Generate/MobileView/Container/InvoicePreview/index.tsx [hunk 2]
```diff
@@ -40,25 +40,31 @@ const InvoicePreviewContainer = ({
     )}
     <InvoiceInfo
       amount={amount}
-      currency={invoiceDetails.companyDetails.currency}
-      dateFormat={invoiceDetails.companyDetails.date_format}
+      currency={currency || invoiceDetails.companyDetails.currency}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Generate/MobileView/Container/InvoicePreview/index.tsx:43]: destructure `invoiceDetails`

File: app/javascript/src/components/Invoices/Generate/MobileView/Container/MenuContainer/InvoiceDetails.tsx [hunk 3]
```diff
@@ -122,20 +122,24 @@ const InvoiceDetails = ({
...
                     defaultValue={null}
-                    options={invoiceDetails.clientList}
                     placeholder="Search"
                     styles={reactSelectStyles.InvoiceDetails}
                     components={{
                       DropdownIndicator,
                       IndicatorSeparator: () => null,
                     }}
+                    options={
+                      invoiceDetails.clientList
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Generate/MobileView/Container/MenuContainer/InvoiceDetails.tsx:139]: destructure `invoiceDetails`

File: app/javascript/src/components/Invoices/Generate/MobileView/Header.tsx [hunk 4]
```diff
@@ -6,11 +6,23 @@ import { Button } from "StyledComponents";
...
+    if (activeSection == sections.generateInvoice || sections.invoicePreview) {
+      return isEdit ? "Edit Invoice" : activeSection.label;
+    }
+
+    return activeSection.label;
+  };
+
   const handleBackAction = () => {
+    if (isEdit) {
+      navigate(-1);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Generate/MobileView/Header.tsx:23]: I hope the user wont navigate to any other screen from edit. If not then this back button wont work properly

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Invoices/Edit/index.tsx (hunk 1 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/Invoices/Edit/index.tsx:161]: cannot destructure `invoicedetails ` on this page, as the state names and field names from `invoicedetails` objects are the same.

File: app/javascript/src/components/Invoices/Generate/MobileView/Header.tsx (hunk 4 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/Invoices/Generate/MobileView/Header.tsx:23]: Handled the other conditions.

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**